internal/network/manifests_generator.go

```diff
@@ -238,6 +241,138 @@ func (m *ManifestsGenerator) AddSchedulableMastersManifest(ctx context.Context,
 	return nil
 }
 
+const diskEncryptionManifest = `apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  name: {{ .ROLE }}-{{ .MODE }}
+  labels:
+    machineconfiguration.openshift.io/role: {{ .ROLE }}
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      luks:
+        - name: root
+          device: /dev/disk/by-partlabel/root
+          clevis:
+		  {{- if eq .MODE "tpm" }}
+            tpm2: true
+		  {{- else if eq .MODE "tang" }}
+            tang:{{ .TANG_INFO }}
+		  {{- end }}
+          options: [--cipher, aes-cbc-essiv:sha256]
+          wipeVolume: true
+      filesystems:
+        - device: /dev/mapper/root
+          format: xfs
+          wipeFilesystem: true
+          label: root
+{{- if eq .MODE "tang" }}
+  kernelArguments:
+    - rd.neednet=1
+{{- end }}`
+
+func (m *ManifestsGenerator) createDiskEncryptionManifest(ctx context.Context, log logrus.FieldLogger, c *common.Cluster,
+	manifestParams map[string]string, filename string) error {
+
+	log.Infof("Creating manifest to encrypt installation disk on %s nodes using %s encryption", manifestParams["ROLE"], manifestParams["ROLE"])
+
+	content, err := fillTemplate(manifestParams, diskEncryptionManifest, log)
+	if err != nil {
+		log.WithError(err).Error("Failed to parse disk encryption manifest's template")
+		return err
+	}
+
+	if err := m.createManifests(ctx, c, filename, content); err != nil {
+		log.WithError(err).Errorf("Failed to create manifest to encrypt installation disk")
+		return err
+	}
+
+	return nil
+}
+
+func (m *ManifestsGenerator) AddDiskEncryptionManifest(ctx context.Context, log logrus.FieldLogger, c *common.Cluster) error {
+
+	if reflect.DeepEqual(c.DiskEncryption, &models.DiskEncryption{}) {
+		return nil
+	}
+
+	manifestParams := map[string]string{}
+
+	switch c.DiskEncryption.Mode {
+
+	case models.DiskEncryptionModeTpmv2:
+
+		manifestParams["MODE"] = "tpm"
+
+	case models.DiskEncryptionModeTang:
+
+		var tangServers []struct {
+			URL        string
+			Thumbprint string
+		}
+		if err := json.Unmarshal([]byte(c.DiskEncryption.TangServers), &tangServers); err != nil {
+			log.WithError(err).Error("faild to unmarshal tang_server from cluster object")
+			return err
+		}
+		var tangInfoStr string
+		for _, ts := range tangServers {
+			tangInfoStr += fmt.Sprintf("\n%s- url: %s\n%sthumbprint: %s", strings.Repeat(" ", 14), ts.URL, strings.Repeat(" ", 16), ts.Thumbprint)
```

Comment:
It's probably cleaner to use a range function in the template?

Reply:
I wasn't aware of that option :)
Fixing it now.